File: strings/divideAStringIntoGroupsOfSizeK.java
// https://leetcode.com/problems/divide-a-string-into-groups-of-size-k/
package strings;

import java.util.ArrayList;

public class divideAStringIntoGroupsOfSizeK {
    public static void main(String[] args) {

        SolutionDASIGOSK solution = new SolutionDASIGOSK();
        solution.divideString("abcdefghi", 3, 'x');
        solution.divideString("abcdefghij", 3, 'x');
        
    }
}

class SolutionDASIGOSK {
    public String[] divideString(String s, int k, char fill) {

        ArrayList<String> arrayList = new ArrayList<String>();
        
        int lengthOfString = s.length();

        // Filling the string with given fill character if the string is short
        if(lengthOfString%k !=0) s = fillString(s,fill,k-(lengthOfString%k));

        int i = 0;
        while(i < lengthOfString){
            arrayList.add(s.substring(i, i+k));
            i+=k;
        }

        System.out.println(arrayList);
        return arrayList.toArray(new String[arrayList.size()]);
        
    }
    public String fillString (String s , char fill , int fillTimes){

        StringBuilder sb= new StringBuilder(s);
        
        while (fillTimes-->0){
            sb.append(fill);
        }
        return sb.toString();

    }
}

File: strings/divideAStringIntoGroupsOfSizeK.py
# https://leetcode.com/problems/divide-a-string-into-groups-of-size-k/

class Solution(object):
    def divideString(self, s, k, fill):
        """
        :type s: str
        :type k: int
        :type fill: str
        :rtype: List[str]
        """
        
        # we check whether if the string is short and do we need to fill it with fill
        # if the string is short, we add fill to the end of the string
        if len(s)%k !=0 : s+=(k-(len(s)%k)) * fill
            
        # we split the string into k-length pieces
        return[s[i:i+k] for i in range(0,len(s),k)]